client/src/main/java/org/eclipse/hono/client/impl/AbstractHonoClient.java

```diff
@@ -236,7 +243,8 @@ protected static final void setApplicationProperties(final Message msg, final Ma
             final String sourceAddress,
             final ProtonQoS qos,
             final ProtonMessageHandler messageHandler,
-            final Handler<String> closeHook) {
+            final Handler<String> closeHook,
+            final Handler<AsyncResult<Void>> detachHandler) {
```

Comment:
IMHO we should add this as a variant of the existing method with the existing method delegating to the new variant, passing in a `null` detach handler.